pkg/audit/manager.go

```diff
@@ -507,48 +509,10 @@ func (am *Manager) addAuditResponsesToUpdateLists(
 	return nil
 }
 
-func (am *Manager) writeAuditResults(ctx context.Context, resourceList []schema.GroupVersionKind, updateLists map[string][]auditResult, timestamp string, totalViolations map[string]int64) error {
-	// get constraints for each Kind
-	for _, constraintGvk := range resourceList {
-		am.log.Info("constraint", "resource kind", constraintGvk.Kind)
-		instanceList := &unstructured.UnstructuredList{}
-		instanceList.SetGroupVersionKind(constraintGvk)
-		err := am.client.List(ctx, instanceList)
-		if err != nil {
-			return err
-		}
-		am.log.Info("constraint", "count of constraints", len(instanceList.Items))
+func (am *Manager) writeAuditResults(ctx context.Context, resourceList []schema.GroupVersionKind, updateLists map[string][]auditResult, timestamp string, totalViolations map[string]int64) {
+	errChan := make(chan error)
 
-		updateConstraints := make(map[string]unstructured.Unstructured, len(instanceList.Items))
-		// get each constraint
-		for _, item := range instanceList.Items {
-			updateConstraints[item.GetSelfLink()] = item
-		}
-		if len(updateConstraints) > 0 {
-			if am.ucloop != nil {
-				select {
-				case <-am.ucloop.stopped:
-					close(am.ucloop.stop)
-				case <-time.After(time.Duration(*auditInterval) * time.Second):
-					am.log.Info("terminating audit ucloop thread due to auditInterval timeout")
-					close(am.ucloop.stop)
-				}
-			}
-			am.ucloop = &updateConstraintLoop{
-				uc:      updateConstraints,
-				client:  am.client,
-				stop:    make(chan struct{}),
-				stopped: make(chan struct{}),
-				ul:      updateLists,
-				ts:      timestamp,
-				tv:      totalViolations,
-				log:     am.log,
-			}
-			am.log.Info("starting update constraints loop", "updateConstraints", updateConstraints)
-			go am.ucloop.update()
-		}
-	}
-	return nil
+	go am.updateAll(ctx, resourceList, updateLists, timestamp, totalViolations, errChan)
```

Comment:
We need some way to preempt the previous thread writing results so that we don't have two competing processes trying to write to `auditResults`

Basically this code:

https://github.com/open-policy-agent/gatekeeper/blob/3a4b86e415f46ecff431b52af714ecdf42bd4337/pkg/audit/manager.go#L528-L534

Which was correct in principle, but just implemented at the wrong granularity -- it was meant to handle 1 extra goroutine, not N extra goroutines.